src/sentry/tasks/sentry_apps.py

```diff
@@ -102,3 +107,148 @@ def send_alert_event(event, rule, sentry_app_id):
+    # Some resources are named differently than their model. eg. Group vs
+    # Issue. Looks up the human name for the model. Defaults to the model name.
+    name = RESOURCE_RENAMES.get(model.__name__, model.__name__.lower())
+
+    # We may run into a race condition where this task executes before the
+    # transaction that creates the Group has committed.
+    try:
+        instance = model.objects.get(id=instance_id)
+    except model.DoesNotExist as e:
+        # Explicitly requeue the task, so we don't report this to Sentry until
+        # we hit the max number of retries.
+        return current.retry(exc=e)
+
+    event = '{}.{}'.format(name, action)
+
+    if event not in ALLOWED_EVENTS:
+        return
+
+    org = None
+
+    if isinstance(instance, Group):
+        org = instance.organization
+
+    installations = filter(
+        lambda i: event in i.sentry_app.events,
+        org.sentry_app_installations.select_related('sentry_app'),
+    )
+
+    for installation in installations:
+        send_webhooks(installation, event, data=serialize(instance))
+
+
+@instrumented_task(name='sentry.tasks.sentry_apps.installation_webhook', **TASK_OPTIONS)
+@retry(on=(RequestException, ))
+def installation_webhook(installation_id, user_id, *args, **kwargs):
+    from sentry.mediators.sentry_app_installations import InstallationNotifier
+
+    extra = {
+        'installation_id': installation_id,
+        'user_id': user_id,
+    }
+
+    try:
+        install = SentryAppInstallation.objects.get(id=installation_id)
+    except SentryAppInstallation.DoesNotExist:
+        logger.info('installation_webhook.missing_installation', extra=extra)
+        return
+
+    try:
+        user = User.objects.get(id=user_id)
+    except User.DoesNotExist:
+        logger.info('installation_webhook.missing_user', extra=extra)
+        return
+
+    InstallationNotifier.run(
+        install=install,
+        user=user,
+    )
+
+
+@instrumented_task(name='sentry.tasks.sentry_apps.workflow_notification', **TASK_OPTIONS)
+@retry(on=(RequestException, ))
+def workflow_notification(installation_id, issue_id, type, user_id, *args, **kwargs):
+    extra = {
+        'installation_id': installation_id,
+        'issue_id': issue_id,
+    }
+
+    try:
+        install = SentryAppInstallation.objects.get(id=installation_id)
+    except SentryAppInstallation.DoesNotExist:
+        logger.info('workflow_notification.missing_installation', extra=extra)
+        return
+
+    try:
+        issue = Group.objects.get(id=issue_id)
+    except Group.DoesNotExist:
+        logger.info('workflow_notification.missing_issue', extra=extra)
+        return
+
+    user = None
+
+    try:
+        user = User.objects.get(id=user_id) if user_id else None
+    except User.DoesNotExist:
+        logger.info('workflow_notification.missing_user', extra=extra)
+
+    send_webhooks(
+        installation=install,
+        event=u'issue.{}'.format(type),
+        data=serialize(issue),
```

Comment:
So in keeping the data representation consistent, we should have `data['resource']` so that we can add other values. I don't think it'd be too hard to add `resolution_type` since that's already given to us in one of the signals. Maybe we can add it to `resolve_in_commit`.
```
{
    "issue": serialize(issue),
    "resolution_type": <resolution_type>
}
```